101-sql-logical-server/azuredeploy.json

```diff
@@ -130,15 +139,33 @@
           ]
       },
       {
-          "condition": "[parameters('enableADS')]",
+          "condition": "[and(parameters('enableADS'),parameters('enableVA'))]",
           "type": "Microsoft.Storage/storageAccounts",
           "apiVersion": "2019-04-01",
           "location": "[parameters('location')]",
           "name": "[variables('storageName')]",
           "sku": {
               "name": "Standard_LRS"
           },
-          "kind": "StorageV2"
-      }
+          "kind": "StorageV2",
+          "resources": [
+              {
+                    "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+                    "apiVersion": "2018-09-01-preview",
+                    "name": "[concat(variables('storageName'), '/Microsoft.Authorization/', variables('uniqueRoleGuid') )]",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Sql/servers', parameters('serverName'))]",
+                        "[resourceId('Microsoft.Storage/storageAccounts', variables('storageName'))]"
+                    ],
+                    "properties": {
+                        "roleDefinitionId": "[variables('StorageBlobContributor')]",
+                        "principalId": "[reference(resourceId('Microsoft.Sql/servers', parameters('serverName')), '2018-06-01-preview', 'Full').identity.principalId]",
+                        "scope": "[resourceId('Microsoft.Storage/storageAccounts', variables('storageName'))]",
+                        "principalType": "ServicePrincipal"
+                    },
+                    "condition": "[and(parameters('enableADS'),parameters('enableVA'))]"
```

Comment:
move this to the top of the resource definition

Reply:
Why ?

Reply:
We're trying to establish some consistency:

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties